tests/jballerina-unit-test/src/test/resources/test-src/types/constant/constant-expression-negative.bal

```diff
@@ -35,3 +35,22 @@ const int CONST1 =  ~~false; // operator '~' not defined for 'boolean'
 const int CONST2 =  -!false; // operator '-' not defined for 'boolean'
 
 const int CONST3 =  -~false; // operator '~' not defined for 'boolean'
+
+const int CI6 = 1 + 2;
+const float CF1 = 1.0 + 2.0;
+const decimal CD1 = 1.0 + 2.0;
+// const byte CBT1 = 1; // Uncomment after fixing #33889
+// const byte CBT2 = 2;
+// const byte CBT3 = CBT1 + CBT2;
+const boolean CB2 = !true;
+const string CS2 = "1" + "2";
+
+function userDefinedTypeTests() {
+    CI6 ci6 = 4; // expected '3', found 'int'
+    CF1 cf1 = 4.0; // expected '3.0f', found 'float'
+    CD1 cd1 = 4.0; // expected '3.0d', found 'float'
+//  CBT3 cbt3 = 4; // expected '3', found 'int'
+    CB2 cb2 = true; // expected 'false', found 'boolean'
+    CS2 cs2 = "4"; // expected '12', found 'string'
+}
```

Comment:
Similar to https://github.com/ballerina-platform/ballerina-lang/pull/33898/files#diff-f7b80acafb6a5c9b42be1aea1acfa78b59911c59f6a479ac5cc5265b47e1a213R319, shall we add some negative tests with all of these constants as expressions also? i.e., when the constant reference is the RHS expression.

Reply:
Like

```ballerina
const int THREE = 1 + 2;
type Foo 3|4;

Foo f = THREE;
```

and similarly negative tests.